packages/renderer/src/lib/ui/ProviderResultPage.svelte

```diff
@@ -35,6 +39,23 @@ function getIcon(check: ImageCheck): IconDefinition {
       return faCircleMinus;
   }
 }
+
+function getCheckedProviders(providers: ProviderUI[]): string[] {
+  return providers.filter(p => p.checked === undefined || p.checked).map(p => p.info.id);
+}
+
+function onProviderChecked(id: string, checked: boolean) {
+  providers = providers.map(p => {
+    if (p.info.id === id) {
+      p.checked = checked;
+    }
+    return p;
+  });
+}
+
+function getFilteredResults(results: CheckUI[], checkedProviders: string[]): CheckUI[] {
+  return results.filter(r => checkedProviders.includes(r.provider.id));
+}
```

Comment:
it makes me feel that maybe we should have used a store on the renderer side, so when we have results, we update the store or we use a derived storer rather than calling getFilteredResults

and we use `$object` rather than `object` in the loops so it's in reactive mode (--> less code, no subscribe/unsubscribe)